x-pack/plugins/lens/public/heatmap_visualization/chart_component.tsx

```diff
@@ -41,20 +117,28 @@ export const HeatmapComponent: FC<HeatmapRenderProps> = ({
   chartsThemeService,
   onClickValue,
   onSelectRange,
+  paletteService,
 }) => {
   const chartTheme = chartsThemeService.useChartsTheme();
   const isDarkTheme = chartsThemeService.useDarkMode();
 
   const tableId = Object.keys(data.tables)[0];
   const table = data.tables[tableId];
 
+  const paletteParams = args.palette?.params as CustomPaletteState;
+
   const xAxisColumnIndex = table.columns.findIndex((v) => v.id === args.xAccessor);
   const yAxisColumnIndex = table.columns.findIndex((v) => v.id === args.yAccessor);
 
   const xAxisColumn = table.columns[xAxisColumnIndex];
   const yAxisColumn = table.columns[yAxisColumnIndex];
   const valueColumn = table.columns.find((v) => v.id === args.valueAccessor);
 
+  const minMaxByColumnId = useMemo(() => findMinMaxByColumnId([args.valueAccessor!], table), [
```

Comment:
Is it necessary to use the ! operator here and below? Is there some check for this condition actually being true?